### ISSUE TITLE
--help and -H provide very different text

### ISSUE THREAD

**Issue by KantarBruceAdams** [wkhtmltopdf/wkhtmltopdf#4174]
**Description**  
The javascript-delay option is necessary if processing the output of something like [apidoc.js](https://github.com/apidoc). It is not documented in the output of --help (nor is --debug-javascript) leading you to initially think wkhtmltopdf is broken or can't handle this kind of document.

**Comment by KantarBruceAdams** [comment]
I note that the correct text is displayed on https://wkhtmltopdf.org/usage/wkhtmltopdf.txt just not in the --help output on Debian or RHEL. Perhaps this is some kind of build issue?

**Comment by Tomsgu** [comment]
With `-h` you get the minimal help that includes these two lines:
```
 -H, --extended-help                 Display more extensive help, detailing
                                     less common command switches
```
What's the concrete issue here?

**Comment by KantarBruceAdams** [comment]
The issue is that the difference between -H and -h is very subtle and surprising.
I raised this initially as I assumed that an option was undocumented because it appeared in the extended help but not the normal help.
It is a sensible pattern when you have many command line options but it is also rare. The only other example that springs to mind is gcc which uses --help and --help -v. For gcc the output of both is very different whereas for wkhtmltodpf I mistook the output of -H and -h as versions of the same help document. I skipped it looking for --javascript-delay and missed -H and --extended-help. So my fault for not reading properly.

Is there any improvement that might be made? Perhaps an early line of the documentation could distinguish the type of help being displayed and imply that there is more than one. For example:
Rather than:

>  wkhtmltopdf 0.12.5 (with patched qt)
> 
> Synopsis:
>   wkhtmltopdf [GLOBAL OPTION]... [OBJECT]... <output file>
>   

Maybe something like:

> wkhtmltopdf 0.12.5 (with patched qt): basic help (use -H for more)
> 
> Synopsis:
>   wkhtmltopdf [GLOBAL OPTION]... [OBJECT]... <output file>
>   

and

> wkhtmltopdf 0.12.5 (with patched qt): extended help (use -h for less)
> 
> Synopsis:
>   wkhtmltopdf [GLOBAL OPTION]... [OBJECT]... <output file>
>   

This is just idiot proofing but its always possible to come up with better classes of idiot. 
I won't be offtended if you close the ticket instead.
